viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
+        for item in genres:
+            if item not in least_popular:
+                least_popular.append(item)
+            else:
+                most_popular.append(item)
+        return (most_popular[0])
+            
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    unique = []
+    total_movies_list = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    total_movies_list = users_movies + friends_movies
+
+    for item in total_movies_list:
+        if item not in friends_movies:
+            unique.append(item)
+    return unique
+
+def get_friends_unique_watched(user_data):
+    friends_unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in friends_unique:
+                friends_unique.append(item)
+    return friends_unique
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended = []
+    users_movies = []
+    friends_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in recommended:
+                if item["host"] in user_data["subscriptions"]:
+                    recommended.append(item)
+    return recommended
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_genre = []
+    genre_recommended = []
+    users_movies = []
+    friends_movies = []
+    
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+        most_genre.append(movie['genre'])
+
+    if len(users_movies) == 0:
+        return friends_movies
+
+    max(set(most_genre), key=most_genre.count)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in genre_recommended:
+                if item["host"] in user_data["subscriptions"]:
+                    if most_genre[0] in item['genre']:

```

Comment:
This passes the tests, but it's not actually checking for the correct genre.  A subtle change in the tests would reveal this bug.  This works because the most common genre shows up first in the `user_data["watched"]` list in the test which means it's the first one `most_genre` list here, but that's not guaranteed to be the case.  The logic of the solution needs to work for more than just the test case.  Also, `item["genre"]` and `most_genre[0]` are both strings.  The `in` operator will return true if they are equal because it is checking to see if one string is contained within the other, but there is a potential bug here.  What if we had genres with similar names (ie, "Historical" and "Historical Drama").  This test would return true if the movie genre was "Historical" and the most frequent genre was "Historical Drama".  To test equality of strings use `==`.